### PR TITLE
geant4: patch typo in wroot

### PR DESCRIPTION
**Pull request by DingXuefeng**
Hello again. There is a typo in geant4. Quite weird that it's not cathced for such a long time.

@drbenmorgan  @sethrj

### CODE REVIEW

**Comment by DingXuefeng** [comment]
I also submitted a bug report for geant4, see https://bugzilla-geant4.kek.jp/show_bug.cgi?id=2640

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 1]
```diff
@@ -215,6 +215,7 @@ def std_when(values):
     # As released, 10.0.4 has inconsistently capitalised filenames
     # in the cmake files; this patch also enables cxxstd 14
     patch("geant4-10.0.4.patch", when="@10.0.4")
+    patch("columns.patch", when="@10.4:")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/geant4/package.py]: Prefix with brief comment explaining what the patch does. Link to bugzilla report. Close version range assuming fixed in next release.

**Comment by sethrj** [comment]
I guess no one has compiled this code before? Perhaps it's better if you don't use it...

**Comment by wdconinc** [comment]
> I guess no one has compiled this code before? Perhaps it's better if you don't use it...

This is similar to the veccore fVal bugfix in 0.8.2, a typo in an inlined icc file. If that header is simply installed without any compilation or testing, it's not really guaranteed to be working until some user stumbles on it. Pitfalls of icc files...

**Comment by drbenmorgan** [comment]
It seems to be a change in how compilers treat "present but never used" code - it's a clear error if compiled, but the latest clang/gcc now seem to consider it this (which is good!).

Note this _is_ fixed in the just-released Geant4 v11.3.0. I'd suggest we get that in first, then this can be rebased so there's a correct version range to apply the patch to. That fix will be backported to at least the next patch to 11.2 (early 2025).

**Comment by wdconinc** [comment]
> Note this _is_ fixed in the just-released Geant4 v11.3.0. I'd suggest we get that in first, then this can be rebased so there's a correct version range to apply the patch to. That fix will be backported to at least the next patch to 11.2 (early 2025).

Opened #47961 for v11.3.0.

**Comment by wdconinc** [comment]
> Note this _is_ fixed in the just-released Geant4 v11.3.0. I'd suggest we get that in first, then this can be rebased so there's a correct version range to apply the patch to.

Rebased.

File: var/spack/repos/builtin/packages/geant4/package.py (hunk 1 above)
Comment (wdconinc) [reply var/spack/repos/builtin/packages/geant4/package.py]:
```suggestion
    # Fix member field typo in g4tools wroot
    # See https://bugzilla-geant4.kek.jp/show_bug.cgi?id=2640
    patch("columns.patch", when="@10.4:11.2.2")
```

**Review by sethrj — APPROVED** [review]

**Review by wdconinc — APPROVED** [review]

**Pull request merged**